src/webgpu/api/operation/render_pipeline/sample_mask.spec.ts

```diff
@@ -517,3 +591,229 @@ textureLoad each sample index from the texture and write to a storage buffer to
+    const stencilResultPromises = [];
+
+    for (const alpha0 of alpha0ParamsArray) {
+      alphaValues[0] = alpha0;
+      alphaValues[1] = alpha1;
+      t.device.queue.writeBuffer(alphaValueUniformBuffer, 0, alphaValues);
+
+      const { color, depthStencil } = t.GetTargetTexture(
+        sampleCount,
+        rasterizationMask,
+        pipeline,
+        alphaValueUniformBuffer,
+        2
+      );
+
+      const colorBuffer = t.copySinglePixelTextureToBufferUsingComputePass(
+        TypeF32, // correspond to 'rgba8unorm' format
+        4,
+        color.createView(),
+        sampleCount
+      );
+      const colorResult = t.readGPUBufferRangeTyped(colorBuffer, {
+        type: Float32Array,
+        typedLength: colorBuffer.size / Float32Array.BYTES_PER_ELEMENT,
+      });
+      colorResultPromises.push(colorResult);
+
+      const depthBuffer = t.copySinglePixelTextureToBufferUsingComputePass(
+        TypeF32, // correspond to 'depth24plus-stencil8' format
+        1,
+        depthStencil.createView({ aspect: 'depth-only' }),
+        sampleCount
+      );
+      const depthResult = t.readGPUBufferRangeTyped(depthBuffer, {
+        type: Float32Array,
+        typedLength: depthBuffer.size / Float32Array.BYTES_PER_ELEMENT,
+      });
+      depthResultPromises.push(depthResult);
+
+      const stencilBuffer = t.copySinglePixelTextureToBufferUsingComputePass(
+        TypeU32, // correspond to 'depth24plus-stencil8' format
+        1,
+        depthStencil.createView({ aspect: 'stencil-only' }),
+        sampleCount
+      );
+      const stencilResult = t.readGPUBufferRangeTyped(stencilBuffer, {
+        type: Uint32Array,
+        typedLength: stencilBuffer.size / Uint32Array.BYTES_PER_ELEMENT,
+      });
+      stencilResultPromises.push(stencilResult);
+    }
+
+    const resultsArray = await Promise.all([
+      Promise.all(colorResultPromises),
+      Promise.all(depthResultPromises),
+      Promise.all(stencilResultPromises),
+    ]);
+
+    const checkResults = (
+      results: { data: Float32Array | Uint32Array; cleanup(): void }[],
+      getExpectedDataFn: (
+        sampleCount: number,
+        rasterizationMask: number,
+        sampleMask: number,
+        alphaToCoverageMask: number
+      ) => Float32Array | Uint32Array,
+      // Alpha to coverage mask should be non-decreasing as the alpha value goes up
+      // Result value of color and stencil is in positive correlation to alpha
+      // Result value of depth is in negative correlation to alpha
+      positiveCorrelation: boolean
+    ) => {
+      for (let i = 0; i < results.length; i++) {
+        const result = results[i];
+        const alpha0 = alpha0ParamsArray[i];
+
+        if (alpha0 <= 0) {
+          const expected = getExpectedDataFn(sampleCount, rasterizationMask, sampleMask, 0x0);
+          const check = checkElementsEqual(result.data, expected);
+          t.expect(check === undefined, `alpha <= 0 should have alphaToCoverageMask be 0x0`);
```

Comment:
nit: `t.expectOK(check)` is the canonical way to check these results, although unfortunately it doesn't currently allow you to provide an extra message string